src/calc.y

```diff
@@ -0,0 +1,32 @@
+%start Expr
+%%
+Expr -> Result<u64, ()>:
+      Expr '+' Term { Ok($1? + $3?) }
+    | Term { $1 }
+    ;
+
+Term -> Result<u64, ()>:
+      Term '*' Factor { Ok($1? * $3?) }
+    | Factor { $1 }
+    ;
+
+Factor -> Result<u64, ()>:
+      '(' Expr ')' { $2 }
+    | 'INT'
+      {
+          let v = $1.map_err(|_| ())?;
+          parse_int($lexer.span_str(v.span()))
+      }
+    ;
+%%
+// Any functions here are in scope for all the grammar actions above.
+
+fn parse_int(s: &str) -> Result<u64, ()> {
+    match s.parse::<u64>() {
+        Ok(val) => Ok(val),
+        Err(_) => {
+            eprintln!("{} cannot be represented as a u64", s);
```

Comment:
Does it make sense to print an error and then propagate it?

It you only propagate it (maybe using your own error type?) will main print it when it bubbles up to the top-level?

Reply:
Custom error added and its propagation but I am not sure how to access the error at runtime :no_mouth:...

When I run it with non-numeric input such as "a" , I get the evaluated result from `parse` function as None (containing none of the error information propagated from `parse_int` function). 

I guess its caused `calc_y::parse(&lexer)` to return None if the operation failed?

Am I doing something weird?
![image](https://user-images.githubusercontent.com/6065236/217867626-f1f451d2-f3c6-4985-8586-0cfdd27ecaff.png)

Reply:
Is this still an issue we need to fix? Or did your later commits sort this?